Content.Client/Chat/ChatManager.cs

```diff
@@ -33,37 +49,62 @@ public void SetChatBox(ChatBox chatBox)
             if (_currentChatBox != null)
             {
                 _currentChatBox.TextSubmitted -= _onChatBoxTextSubmitted;
+                _currentChatBox.FilterPressed -= _onFilterButtonToggled;
             }
 
             _currentChatBox = chatBox;
             if (_currentChatBox != null)
             {
                 _currentChatBox.TextSubmitted += _onChatBoxTextSubmitted;
+                _currentChatBox.FilterPressed += _onFilterButtonToggled;
             }
         }
 
         private void _onChatMessage(MsgChatMessage message)
         {
             Logger.Debug($"{message.Channel}: {message.Message}");
 
-            var color = Color.DarkGray;
-            var messageText = message.Message;
-            if (!string.IsNullOrEmpty(message.MessageWrap))
+            if (filteredHistory.ContainsValue(message))
```

Comment:
This check seems pretty redundant. I can't see any case in which it *should* pass and it has some pretty bad performance complexity. Just remove it.

Reply:
Removed!